File: solutions/day04/solution.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let i = 0;
    let j = height.length -1;
    let maxArea = 0;
    while(i<j){
        let minH = height[i] < height[j] ? height[i] : height[j];
        let area = minH * (j - i);
        if(maxArea < area) {
            maxArea = area;
        }
        if(height[i] < height[j]){
            i++;
        } else {
            j--;
        }
    }

    return maxArea;
};

File: solutions/day04/solution.java
class Solution {
    public int maxArea(int[] height) {
       int area=0;
       int left=0,right=height.length-1;
       int maxi=0;
       while(left<right){
            area=Math.min(height[left],height[right])*(right-left);
            maxi=Math.max(maxi,area);
            if(height[left]>height[right]){
                right--;
            }
            else{
                left++;
            }
       }
       return maxi;
    }
}

